marlowe-cli/src/Language/Marlowe/CLI/Command.hs

```diff
@@ -115,25 +120,33 @@ parseCommand networkId socketPath version =
     (
           O.helper
       <*> versionOption version
-      <*> (noEraParser <|> alonzoParser <|> babbageParser)
+      <*> someCommandParser
     )
     (
          O.fullDesc
       <> O.progDesc "Utilities for Marlowe."
       <> O.header "marlowe-cli : a command-line tool for Marlowe contracts"
     )
   where
-    alonzoParser = SomeCommand <$> alonzoFlagParser <*> commandParser
-    babbageParser = SomeCommand <$> babbageFlagParser <*> commandParser
-    noEraParser = SomeCommand ScriptDataInBabbageEra <$> commandParser
-    alonzoFlagParser = O.flag ScriptDataInAlonzoEra ScriptDataInAlonzoEra $ mconcat
-      [ O.long "alonzo-era"
-      , O.help "Read and write Alonzo transactions"
-      ]
-    babbageFlagParser = O.flag' ScriptDataInBabbageEra $ mconcat
-      [ O.long "babbage-era"
-      , O.help "Read and write Babbage transactions"
-      ]
+    eraOption :: O.Parser SomeEra
+    eraOption =
+      asum
+        [
+           O.flag' (SomeEra ScriptDataInAlonzoEra)
+             (  O.long "alonzo-era"
+             <> O.help "Specify the Alonzo era"
+             )
+         , O.flag' (SomeEra ScriptDataInBabbageEra)
+             (  O.long "babbage-era"
+             <> O.help "Specify the Babbage era (default)"
+             )
+         , pure (SomeEra ScriptDataInBabbageEra)
+         ]
+    someCommandParser :: O.Parser SomeCommand
+    someCommandParser = O.BindP eraOption mkSomeCommandParser
```

Comment:
Oh, I didn't know there was a `BindP`... wonder how come there isn't a `Monad` instance for `Parser` then :thinking:

Reply:
I didn't either, but the package is so focused on applicative that it doesn't offer a monadic escape hatch.